idaes/apps/grid_integration/multiperiod/price_taker_model.py

```diff
@@ -91,11 +80,30 @@ def horizon_length(self, value):
         if value <= 0:
             raise ValueError(f"horizon_length must be > 0, but {value} is provided.")
         if not isinstance(value, int):
-            raise ValueError(
+            raise TypeError(
                 f"horizon_length must be an integer, but {value} is not an integer"
             )
         self._horizon_length = value
 
+    @staticmethod
```

Comment:
Could you comment on why this is a `staticmethod`?

Also, a minor technical comment but for the doc string this should be called a method, not a function. E.g. https://www.codecademy.com/article/fwd-js-methods-functions

Reply:
If I don't add the `staticmethod` label, I see the following message `Method 'compute_sse' may be 'static'`, so I added it just to be safe.

Reply:
That is not a good answer - the better answer is because it does not make use of the class itself (i.e., the `self` argument), but that leads into a deeper design question of what this method is intended to be. One question that came to mind as I read this again was whether this is intended to be something a user will interact with or not.

If a user is not expected to directly interact with this, it might be worth considering making this a private method as well (so as not to clutter the docs).

Reply:
I see no reason why a user would need to interaction with this. 

Does a private method essentially just mean the docstring is restructured as done in the following example:
`    def _set_scaling_factor(
        self, component, component_type, scaling_factor, overwrite=None
    ):
        """
        PRIVATE METHOD

        The purpose of this method is to apply the correct max and min limits to scaling factors
        before setting them. Which set of limits to apply is determined solely by the component_type
        argument, and no type checking is performed.
        """`